### ISSUE TITLE
`nodejs` FTBFS with Python 3.12

### ISSUE THREAD

**Issue by mhmdanas** [void-linux/void-packages#46592]
### Is this a new report?

Yes

### System Info

Void 6.5.6_1 x86_64 GenuineIntel uptodate rrrmFFFFFFFF

### Package(s) Affected

nodejs-18.16.0_3

### Does a report exist for this bug with the project's home (upstream) and/or another distro?

https://github.com/nodejs/node-gyp/issues/2869 seems related.

### Expected behaviour

The template builds successfully.

### Actual behaviour

The template fails to configure:

```
=> nodejs-18.16.0_3: running do_configure ...
Please use python3.11 or python3.10 or python3.9 or python3.8 or python3.7 or python3.6.
Node.js configure: Found Python 3.12.0...
```

### Steps to reproduce

1. `./xbps-src pkg nodejs`

**Comment by ahesford** [comment]
Adding `python3-setuptools` to `hostmakedepends` and patching `configure` to not reject Python 3.12 outright at least starts the build. I'm running some build tests and will push the patch if everything succeeds.

**Comment by ahesford** [comment]
Seems to be fixed in d5a68be7f3777ffd87ea0b3bb8bcea24374a6d18
